runtime/src/registry/mod.rs

```diff
@@ -0,0 +1,264 @@
+                                             asset_info,
+                                             mint_info)?;
+
+            // Mint event
+            Self::deposit_event(RawEvent::Mint(registry_id, token_id));
+
+            Ok(())
+        }
+    }
+}
+
+// Auxillary methods of the module for internal use
+impl<T: Trait> Module<T> {
+    fn get_document_root(anchor_id: T::Hash) -> Result<H256, dispatch::DispatchError> {
+        let root = match <anchor::Module<T>>::get_anchor_by_id(anchor_id) {
+            Some(anchor_data) => Ok(anchor_data.doc_root),
+            None => Err(Error::<T>::DocumentNotAnchored),
+        }?;
+
+        Ok( H256::from_slice(root.as_ref()) )
+    }
+
+    fn create_new_registry_id() -> Result<RegistryId, dispatch::DispatchError> {
+        let id = <RegistryNonce>::get();
+
+        // TODO: Make a U160 type for RegistryId with the uint crate.
+        // Passing through U256 is inefficient and H160 is unneeded.
+        let mut res = Vec::<u8>::with_capacity(32);
+        unsafe { res.set_len(32); }
+        // U256 > H160 so no need for a checked_add
+        U256::from_big_endian(id.as_bytes())
+             .saturating_add(U256::one())
+             .to_big_endian(&mut res);
+
+        // Interpreted in big endian
+        let nplus1 = H160::from_slice(&res[0..20]);
+
+        // Update the nonce
+        <RegistryNonce>::put( nplus1 );
+
+        Ok(id)
+    }
+}
+
+// Implement the verifier registry. This module verifies data fields that are custom defined
+// by a registry and provided in the MintInfo during a mint invocation.
+impl<T: Trait> VerifierRegistry for Module<T> {
+    type AccountId    = <T as frame_system::Trait>::AccountId;
+    type RegistryId   = RegistryId;
+    type RegistryInfo = RegistryInfo;
+    type AssetId      = AssetId;
+    type AssetInfo    = <T as nft::Trait>::AssetInfo;
+    type MintInfo     = MintInfo<<T as frame_system::Trait>::Hash, H256>;
+
+    // Registries with identical RegistryInfo may exist
+    fn create_registry(mut info: Self::RegistryInfo) -> Result<Self::RegistryId, dispatch::DispatchError> {
+        // Generate registry id as nonce
+        let id = Self::create_new_registry_id()?;
+
+        // Create a field of the registry that is the registry id encoded with a prefix
+        let pre_reg = [NFTS_PREFIX, id.as_bytes()].concat();
+        info.fields.push(pre_reg);
+
+        // Insert registry in storage
+        Registries::insert(id.clone(), info);
+
+        Ok(id)
+    }
+
+    fn mint(caller: &<T as frame_system::Trait>::AccountId,
+            owner_account: &<T as frame_system::Trait>::AccountId,
+            asset_id: &Self::AssetId,
+            asset_info: T::AssetInfo,
+            mint_info: MintInfo<<T as frame_system::Trait>::Hash, H256>,
+    ) -> Result<(), dispatch::DispatchError> {
+        let (registry_id, token_id) = AssetIdRef::from(asset_id).destruct();
+        let registry_info = Registries::get(registry_id);
```

Comment:
Wouldn't this be None if registry_id doesnt exist? In which case the below check can be replace by is_ok_or(RegistryDoesNotExist).

Reply:
I've been meaning to look deeper into this. For some reason, the StorageMap interface does not provide a try_get() - only get - [frame_support/storage/trait.StorageMap](https://substrate.dev/rustdocs/v2.0.0-rc6/frame_support/storage/trait.StorageMap.html).

I believe if a value doesn't exist, the default is returned (which is why a value must impl Default). Though I'm having trouble finding proof beyond [this point in the knowledge base](https://substrate.dev/docs/en/knowledgebase/runtime/storage#default-values).

It may be faster to compare the result to the default value but `contains_key` feels safer until I know for sure about the default values.